src/gatherers/critical-network-chains.js

```diff
@@ -0,0 +1,164 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const Gather = require('./gather');
+const childProcess = require('child_process');
+const fs = require('fs');
+const log = require('../lib/log.js');
+
+const flatten = arr => arr.reduce((a, b) => a.concat(b), []);
+const contains = (arr, elm) => arr.indexOf(elm) > -1;
+
+class Node {
+  get requestId() {
+    return this.request.requestId;
+  }
+  constructor(request, parent) {
+    this.children = [];
+    this.parent = parent;
+    this.request = request;
+  }
+
+  setParent(parentNode) {
+    this.parent = parentNode;
+  }
+
+  addChild(childNode) {
+    this.children.push(childNode);
+  }
+
+}
+
+class CriticalNetworkChains extends Gather {
+
+  get criticalPriorities() {
```

Comment:
Let's link to https://docs.google.com/document/u/1/d/1bCDuq9H1ih9iNjgzyAL0gpwNFiEP4TZS-YLRp_RuMlc/edit to explain this.